zsh-copilot.plugin.zsh

```diff
@@ -41,55 +45,70 @@ fi
             $SYSTEM"
     fi
+
     # Get input
     local input=$(echo "${BUFFER:0:$CURSOR}" | tr '\n' ';')
     input=$(echo "$input" | sed 's/"/\\"/g')
 
     _zsh_autosuggest_clear
     zle -R "Thinking..."
 
-    PROMPT=$(echo "$PROMPT" | tr -d '\n')
-    # Wasn't able to get this to work :(
-    # if [[ "$ZSH_COPILOT_SEND_GIT_DIFF" == 'true' ]]; then
-    #     if [[ $(git rev-parse --is-inside-work-tree) == 'true' ]]; then
-    #         local git_diff=$(git diff --staged --no-color)
-    #         local git_exit_code=$?
-    #         git_diff=$(echo "$git_diff" | tr '\\' ' ' | sed 's/[\$\"\`]/\\&/g' | tr '\\' '\\\\' | tr -d '\n')
-    #
-    #         if [[ git_exit_code -eq 0 ]]; then
-    #             PROMPT="$PROMPT; This is the git diff: <---->$git_diff<----> You may provide a git commit message if the user is trying to commit changes. You are an expert at committing changes, you don't give generic messages. You give the best commit messages"
-    #         fi
-    #     fi
-    # fi
-
-    local data="{
-            \"model\": \"gpt-4\",
+    local full_prompt=$(echo "$SYSTEM_PROMPT $context_info" | tr -d '\n')
+
+    local data
+    local response
+
+    if [[ "$ZSH_COPILOT_AI_PROVIDER" == "openai" ]]; then
+        data="{
+            \"model\": \"gpt-4o-mini\",
             \"messages\": [
                 {
                     \"role\": \"system\",
-                    \"content\": \"$PROMPT\"
+                    \"content\": \"$full_prompt\"
                 },
                 {
                     \"role\": \"user\",
                     \"content\": \"$input\"
                 }
             ]
         }"
-    local response=$(curl "https://${OPENAI_API_URL}/v1/chat/completions" \
-        --silent \
-        -H "Content-Type: application/json" \
-        -H "Authorization: Bearer $OPENAI_API_KEY" \
-        -d $data)
-    local message=$(echo "$response" | jq -r '.choices[0].message.content')
-
-    # zle -U "$suggestion"
+        response=$(curl "https://${OPENAI_API_URL}/v1/chat/completions" \
+            --silent \
+            -H "Content-Type: application/json" \
+            -H "Authorization: Bearer $OPENAI_API_KEY" \
+            -d "$data")
+        local message=$(echo "$response" | jq -r '.choices[0].message.content')
+    elif [[ "$ZSH_COPILOT_AI_PROVIDER" == "anthropic" ]]; then
+        data="{
+            \"model\": \"claude-3-5-sonnet-20240620\",
+            \"max_tokens\": 1000,
+            \"system\": \"$full_prompt\",
+            \"messages\": [
+                {
+                    \"role\": \"user\",
+                    \"content\": \"$input\"
+                }
+            ]
+        }"
+        response=$(curl "https://${ANTHROPIC_API_URL}/v1/messages" \
+            --silent \
+            -H "Content-Type: application/json" \
+            -H "x-api-key: $ANTHROPIC_API_KEY" \
+            -H "anthropic-version: 2023-06-01" \
```

Comment:
Is this date correct? I don't know whow Anthropic works but this seems kinda old to me

Reply:
![image](https://github.com/user-attachments/assets/25511608-7af6-45d1-8b91-2f362446dab0)

Reply:
Can you send me the link to that page?

Reply:
https://docs.anthropic.com/en/api/messages